From 263e09bfd35e71591424dfe83e7561c68eee25c5 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Wed, 20 Feb 2013 16:57:39 +0800
Subject: [PATCH 07/25] ENGR00251059 support 2 channel directoutput

1. fix bug in android for sup_channel and sup_format
2. support 2 channel directoutput

Signed-off-by: b02247 <b02247@freescale.com>
---
 audio/AudioPolicyManagerBase.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 8cf7523..0f2cce6 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2009 The Android Open Source Project
- * Copyright (C) 2012 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -507,6 +507,10 @@ AudioPolicyManagerBase::IOProfile *AudioPolicyManagerBase::getProfileForDirectOu
                                                                audio_channel_mask_t channelMask,
                                                                audio_output_flags_t flags)
 {
+
+    if (channelMask == AUDIO_CHANNEL_OUT_STEREO && flags != AUDIO_OUTPUT_FLAG_DIRECT)
+        return 0;
+
     for (size_t i = 0; i < mHwModules.size(); i++) {
         if (mHwModules[i]->mHandle == 0) {
             continue;
@@ -788,7 +792,6 @@ status_t AudioPolicyManagerBase::startOutput(audio_io_handle_t output,
         if (isInCall()) {
             handleIncallSonification(stream, true, false);
         }
-
         // apply volume rules for current stream and device if necessary
         checkAndSetVolume(stream,
                           mStreams[stream].getVolumeIndex(newDevice),
-- 
1.8.0

